File: BMICalculator/app/src/main/java/com/example/calorie/CalorieRecommendation.java
package com.example.calorie;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.Spinner;

import com.example.bmicalculator.R;

public class CalorieRecommendation extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calorie_recommendation);

        Spinner spinner = (Spinner) findViewById(R.id.gender);
        // Create an ArrayAdapter using the string array and a default spinner layout
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this,
                R.array.spinner, android.R.layout.simple_spinner_item);
        // Specify the layout to use when the list of choices appears
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        // Apply the adapter to the spinner
        spinner.setAdapter(adapter);

        EditText height = (EditText) this.findViewById(R.id.height);
        EditText weight = (EditText) this.findViewById(R.id.weight);
        EditText age = (EditText) this.findViewById(R.id.age);
        Spinner gender = (Spinner) this.findViewById(R.id.gender);

        if (gender.getSelectedItem().toString().equals("Female")) {
            Float calorie = 655.1 + (9.563 * weight)+(1.85 * height)-(4.676 * age);
        } else {
            Float calorie = 66.5 + (13.75 * weight)+(5.003 * height)-(6.775 * age);
        }

    }
}

File: BMICalculator/app/src/main/java/com/example/calorie/CalorieRecommendation.kt
package com.example.calorie

import androidx.appcompat.app.AppCompatActivity
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import android.os.Bundle
import android.view.View
import com.example.bmicalculator.R
import android.widget.ArrayAdapter
import kotlin.jvm.internal.Intrinsics

class CalorieRecommendation : AppCompatActivity() {
    private var height: EditText? = null
    private var weight: EditText? = null
    private var age: EditText? = null
    private var gender: Spinner? = null
    private var result: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calorie_recommendation)
        val spinner = findViewById<View>(R.id.gender) as Spinner
        // Create an ArrayAdapter using the string array and a default spinner layout
        val adapter = ArrayAdapter.createFromResource(
            this,
            R.array.spinner, android.R.layout.simple_spinner_item
        )
        // Specify the layout to use when the list of choices appears
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        // Apply the adapter to the spinner
        spinner.adapter = adapter
        height = findViewById<View>(R.id.height) as EditText
        weight = findViewById<View>(R.id.weight) as EditText
        age = findViewById<View>(R.id.age) as EditText
        gender = findViewById<View>(R.id.gender) as Spinner
        result = findViewById<View>(R.id.result) as TextView
    }

    fun calculateCalories(v: View) {
        var calorie = 0.0
        calorie = if (gender!!.selectedItem.toString() == "Female") {
            655.1 + 9.563 * weight!!.text.toString().toFloat() + 1.85 * height!!.text.toString()
                .toFloat() - 4.676 * age!!.text.toString().toFloat()
        } else {
            66.5 + 13.75 * weight!!.text.toString().toFloat() + 5.003 * height!!.text.toString()
                .toFloat() - 6.775 * age!!.text.toString().toFloat()
        }
        displayText(calorie)
    }

    private fun displayText(calories: Double) {
        val bmiLabel = calories.toString() + ""
        val result = result
        Intrinsics.checkNotNull(result)
        result!!.text = bmiLabel
    }
}